vpr/src/route/route_timing.h

```diff
@@ -10,9 +10,196 @@
+    // empty).  When using cost_params.astar_fac = 0, for efficiency the
+    // RouterLookahead used should be the NoOpLookahead.
+    std::vector<t_heap> timing_driven_find_all_shortest_paths_from_route_tree(
+        t_rt_node* rt_root,
+        const t_conn_cost_params cost_params,
+        t_bb bounding_box,
+        RouterStats& router_stats);
+
+  private:
+    // Mark that inode has been modified, and needs to be reset in
+    // reset_path_costs.
+    void add_to_mod_list(int inode) {
+        if (std::isinf(rr_node_route_inf_[inode].path_cost)) {
+            modified_rr_node_inf_.push_back(inode);
+        }
+    }
+
+    // Update the route path to the node pointed to by cheapest.
+    inline void update_cheapest(t_heap* cheapest) {
+        update_cheapest(cheapest, &rr_node_route_inf_[cheapest->index]);
+    }
+
+    inline void update_cheapest(t_heap* cheapest, t_rr_node_route_inf* route_inf) {
+        //Record final link to target
+        add_to_mod_list(cheapest->index);
+
+        route_inf->prev_node = cheapest->u.prev.node;
+        route_inf->prev_edge = cheapest->u.prev.edge;
+        route_inf->path_cost = cheapest->cost;
+        route_inf->backward_path_cost = cheapest->backward_path_cost;
+    }
+
+    // Common logic from timing_driven_route_connection_from_route_tree and
+    // timing_driven_route_connection_from_route_tree_high_fanout for running
+    // connection router.
+    t_heap* timing_driven_route_connection_common_setup(
+        t_rt_node* rt_root,
+        int sink_node,
+        const t_conn_cost_params cost_params,
+        t_bb bounding_box);
+
+    // Finds a path to sink_node, starting from the elements currently in the
+    // heap.
+    //
+    // This is the core maze routing routine.
+    //
+    // Note: For understanding the connection router, start here.
+    //
+    // Returns either the last element of the path, or nullptr if no path is
+    // found
+    t_heap* timing_driven_route_connection_from_heap(
+        int sink_node,
+        const t_conn_cost_params cost_params,
+        t_bb bounding_box);
+
+    // Expand this current node if it is a cheaper path.
+    void timing_driven_expand_cheapest(
+        t_heap* cheapest,
+        int target_node,
+        const t_conn_cost_params cost_params,
+        t_bb bounding_box);
+
+    // Expand each neighbor of the current node.
+    void timing_driven_expand_neighbours(
+        t_heap* current,
+        const t_conn_cost_params cost_params,
+        t_bb bounding_box,
+        int target_node);
+
+    // Conditionally adds to_node to the router heap (via path from from_node
+    // via from_edge).
+    //
+    // RR nodes outside bounding box specified in bounding_box are not added
+    // to the heap.
+    void timing_driven_expand_neighbour(
+        t_heap* current,
+        const int from_node,
+        const RREdgeId from_edge,
+        const t_edge_size from_edge_int,
```

Comment:
Suggest calling this one "from_node_edge_idx"

Reply:
done